stubs/Pygments/pygments/__init__.pyi

```diff
@@ -1,5 +1,16 @@
-from typing import Any
+from io import FileIO
+from typing import Any, TypeVar, overload
+
+from pygments.formatters import Formatter
+
+_T = TypeVar("_T", str, bytes)
 
 def lex(code, lexer): ...
-def format(tokens, formatter, outfile: Any | None = ...): ...
-def highlight(code, lexer, formatter, outfile: Any | None = ...): ...
+@overload
+def format(tokens, formatter: Formatter[Any], outfile: FileIO) -> None: ...
```

Comment:
The docstring describes outfile as "an object with a `write` method", so we should probably use `_typeshed.SupportsWrite` instead of `FileIO`. This also applies to `highlight()`.